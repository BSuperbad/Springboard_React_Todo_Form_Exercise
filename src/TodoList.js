import React, {useState} from "react";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () =>{
    const INITIAL_STATE = [
        {id: uuid(), todo_text: 'Take out Trash'},
        {id: uuid(), todo_text: 'Do Dishes'}
    ];
const [todos, setTodos] = useState(INITIAL_STATE);
const addNewTodo = (newTodo) => {
    setTodos(todos=> [...todos, {...newTodo, id: uuid()}])
};

const removeTodo = (todoId) => {
    setTodos(todos=> todos.filter((todo)=>todo.id !== todoId))
}

return (
    <div>
        <div>
            <h2>Todo List:</h2>
            {todos.map(({id, todo_text})=> (
                <Todo
                id={id}
                key={id}
                todo_text={todo_text}
                removeTodo={removeTodo}
                />
            ))}
        </div>
        <h3>Add a Task to Your Todo List!</h3>
        <NewTodoForm addNewTodo={addNewTodo}/>
    </div>
)

}

export default TodoList;