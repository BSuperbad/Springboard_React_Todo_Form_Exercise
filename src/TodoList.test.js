import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders TodoList without crashing", function () {
    render(<TodoList />);
});

it("matches TodoList snapshot", function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", function(){
    const {queryByPlaceholderText, queryByText} = render(<TodoList />);
    const input = queryByPlaceholderText("Enter a Task");
    const btn = queryByText("Add Task to Todo List");
    expect(queryByText('Vacuum')).not.toBeInTheDocument();
    fireEvent.change(input, {target: {value: "Vacuum"}});
    fireEvent.click(btn);
    expect(queryByText('Vacuum')).toBeInTheDocument();
});