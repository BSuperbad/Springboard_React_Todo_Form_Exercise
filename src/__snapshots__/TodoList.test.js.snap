// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches TodoList snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Todo List:
      </h2>
      <div>
        <ul>
          <li>
            Take out Trash 
            <button>
              X
            </button>
          </li>
        </ul>
      </div>
      <div>
        <ul>
          <li>
            Do Dishes 
            <button>
              X
            </button>
          </li>
        </ul>
      </div>
    </div>
    <h3>
      Add a Task to Your Todo List!
    </h3>
    <form>
      <label
        for="todo_text"
      >
        Add New Todo: 
      </label>
      <input
        id="todo_text"
        name="todo_text"
        placeholder="Enter a Task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Task to Todo List
      </button>
    </form>
  </div>
</DocumentFragment>
`;
